src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -305,12 +305,51 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
             return dbResultSet;
         }
 
+        /// <inheritdoc />
+        public override SqlCommand PrepareDbCommand(
+            SqlConnection conn,
+            string sqltext,
+            IDictionary<string, DbConnectionParam> parameters,
+            HttpContext? httpContext,
+            string dataSourceName)
+        {
+            SqlCommand cmd = conn.CreateCommand();
+            cmd.CommandType = CommandType.Text;
+
+            // Add query to send user data from DAB to the underlying database to enable additional security the user might have configured
+            // at the database level.
+            string sessionParamsQuery = GetSessionParamsQuery(httpContext, parameters, dataSourceName);
+
+            cmd.CommandText = sessionParamsQuery + sqltext;
+            if (parameters is not null)
+            {
+                foreach (KeyValuePair<string, DbConnectionParam> parameterEntry in parameters)
+                {
+                    SqlParameter parameter = cmd.CreateParameter();
+                    parameter.ParameterName = parameterEntry.Key;
+                    parameter.Value = parameterEntry.Value.Value ?? DBNull.Value;
+                    PopulateDbTypeForParameter(parameterEntry, parameter);
```

Comment:
I don't understand why you need `PopulateDbTypeForParameter(parameterEntry, parameter);`

`SqlParameter parameter = cmd.CreateParameter(sqlDbType: parameterEntry.Value.SqlDbType);` achieves the same thing because it auto-resolves the `DbType` .

Reply:
This method has some extra checks as well which validates before assigning values. Also it was added in an older PR. Btw, we do have an issue tracking whether we need DbType, now that we are introducing SqlDbType. So this will be touched in that issue to understand better usage.